### PR TITLE
Evaluate for performance between MAM and E2EE

### PR DESCRIPTION
**Pull request by howjmay**

### CODE REVIEW

File: paper.tex [hunk 1]
```diff
@@ -526,6 +526,32 @@ \subsubsection{Messages Publishment}
...
+\end{itemize}
+
+\begin{figure}[!t]
+    \centering
+    \includegraphics[width=3in]{mam_vs_e2ee}
+    \caption{Compare the performance between MAM and E2EE on Raspberry Pi.}
+    \label{fig:mam_vs_e2ee}
+\end{figure}
+
+Turning now to the experimental evidence on Fig.~\ref{fig:mam_vs_e2ee}. A comparison of the two result reveals elapsing time of E2EE is obviously less than elapsing time of MAM which shows the E2EE process presented in the paper is a much more appropriate solution for low-level embedded system to send authenticated streaming data. Next, regarding the trend of the two lines, the performance of E2EE is obviously more steady than MAM in the given range. Moreover, the performance of E2EE is almost under constant time, but MAM takes linear time on encrypting messages.
```
Comment (jserv) [inline paper.tex]: Remove the sentence "Turning now to the experimental evidence"

File: paper.tex [hunk 2]
```diff
@@ -526,6 +526,32 @@ \subsubsection{Messages Publishment}
...
 
+\subsection{MAM versus E2EE}
+What follows is an account of performance comparison between MAM and E2EE for different payload sizes. Time cost for internet communication and PoW are not included, because the following processes starting from sending message to tangle-accelerator are the common steps. Excluding the common steps allows us to compare the performance between MAM and E2EE more clearly. In addition, MAM Channel creation and E2EE asymmetric key pair generation are not included, too. The two excluded steps take much more time than encrypting messages only, so counting these two steps would cause extreme values.
```
Comment (jserv) [inline paper.tex]: 把 time cost 換為 elapsed time

File: paper.tex [hunk 3]
```diff
@@ -526,6 +526,32 @@ \subsubsection{Messages Publishment}
 
 Therefore, transferring MAM operations to brokers while ensuring the profit and privacy of providers through blind signatures can effectively solve performance problems and lower the threshold to participate in such framework. Brokers can be PCs or powerful machines that runs Ethereum client and Tangle-accelerator\cite{TA}. Where Ethereum client is used to interact with Ethereum and Tangle-accelerator is a caching proxy server for IOTA, which can serve thousands of IOTA requests at once without accessing remote IOTA full nodes frequently and provide PoW acceleration. However, MAM operations still cost a considerable time that improving the performance of MAM is an essential issue that needs to be done for the next step.
 
+\subsection{MAM versus E2EE}
```
Comment (jkrvivian) [inline paper.tex]: Isn't this section should be in section Evaluation?

Comment (howjmay) [reply paper.tex]: Under MAM eveluation?

Comment (jkrvivian) [reply paper.tex]: It's ok to open a new subsection.

Comment (howjmay) [reply paper.tex]: Ok

Comment (jserv) [reply paper.tex]: The comparison should be "MAM vs. the delegated MAM."

File: paper.tex [hunk 4]
```diff
@@ -675,6 +675,31 @@ \subsection{Smart Contract}
...
+\end{itemize}
+
+\begin{figure}[!t]
+    \centering
+    \includegraphics[width=3in]{mam_vs_e2ee}
+    \caption{Compare the performance between MAM and E2EE on Raspberry Pi.}
+    \label{fig:mam_vs_e2ee}
+\end{figure}
+
+On Fig.~\ref{fig:mam_vs_e2ee}, a comparison of the two result reveals elapsing time of E2EE is obviously less than elapsing time of MAM which shows the E2EE process presented in the paper is a much more appropriate solution for low-level embedded system to send authenticated streaming data. Next, regarding the trend of the two lines, the performance of E2EE is obviously more steady than MAM in the given range. Moreover, the performance of E2EE is almost under constant time, but MAM takes linear time on encrypting messages.
```
Comment (jkrvivian) [inline paper.tex]: How do we explain the bump of MAM?

File: paper.tex [hunk 5]
```diff
@@ -675,6 +677,31 @@ \subsection{Smart Contract}
...
+\end{itemize}
+
+\begin{figure}[!t]
+    \centering
+    \includegraphics[width=3in]{mam_vs_e2ee}
+    \caption{Compare the performance between MAM and E2EE on Raspberry Pi.}
+    \label{fig:mam_vs_e2ee}
+\end{figure}
+
+On Fig.~\ref{fig:mam_vs_e2ee}, a comparison of the two result reveals elapsing time of E2EE is obviously less than elapsing time of MAM which shows the E2EE process presented in the paper is a much more appropriate solution for low-level embedded system to send authenticated streaming data. Next, regarding the trend of the two lines, the performance of E2EE is obviously more steady than MAM in the given range. The bump of MAM performance around 750 bytes proves this point. Moreover, the performance of E2EE is almost under constant time, but MAM takes linear time on encrypting messages.
```
Comment (jserv) [inline paper.tex]: Use "in the figure"
Reference: https://english.stackexchange.com/questions/27689/usage-of-in-figure-vs-on-figure

**Pull request merged**